### PR TITLE
 -Ztrait-solver=next: stop depending on old solver

### PR DESCRIPTION
**Pull request by lcnr**
removes the final dependencies on the old solver when `-Ztrait-solver=next` is enabled.

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/query/type_op/implied_outlives_bounds.rs [hunk 1]
```diff
@@ -164,19 +166,29 @@ pub fn compute_implied_outlives_bounds_inner<'tcx>(
...
+    let mut implied_bounds = Vec::new();
+    for ty::OutlivesPredicate(a, r_b) in outlives_bounds {
+        match a.unpack() {
+            ty::GenericArgKind::Lifetime(r_a) => {
+                implied_bounds.push(OutlivesBound::RegionSubRegion(r_b, r_a))
+            }
             ty::GenericArgKind::Type(ty_a) => {
-                let ty_a = ocx.infcx.resolve_vars_if_possible(ty_a);
+                let mut ty_a = ocx.infcx.resolve_vars_if_possible(ty_a);
+                // Need to manually normalize in the new solver as `wf::obligations` does not.
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/traits/query/type_op/implied_outlives_bounds.rs:177]: Why isn't that done? I see that you're bailing out in `WfPredicates::normalize`, but not why

Comment (lcnr) [reply compiler/rustc_trait_selection/src/traits/query/type_op/implied_outlives_bounds.rs:177]: extended the comment in `wf.rs`, the reasons are
- the new solver can only deeply normalize if there is no ambiguity
- we never have to normalize obligations in the new solver, but we do have to normalize the inputs to region solving

**Comment by compiler-errors** [comment]
@rust-timer build 3cfc22c1d7b73b6e6f271d624b47ed04b02eb208

**Pull request merged**